Tiled Tiff Converter Plugin - Version 1.0.0

Nicholas-Schaub opened the pull request:
This PR adds in a tiled tiff converter plugin.
The need for this tool is twofold.
First, the tiled tiff conversion plugin used by WIPP-backend will only convert the first image plane, which means only the first channel in the first timepoint in the first z-slice of an image file. The rest of the data will be discarded. This plugin will save each plane as a separate file, meaning no image information is lost.
Second, WIPP currently only converts a handful of images to tiled tiffs (including jpegs and tiffs), while any other file format will be imported as metadata regardless of whether Bioformats supports it. This plugin will convert any Bioformats supported image type into a tiled tiff, except for images that are stored as .csv or .txt. For images uploaded to WIPP that are not jpeg or tiff and are stored as metadata when uploaded, this plugin has an option to convert metadata files into tiled tiffs.

sunnielyu commented on polus-tiledtiff-converter-plugin/Dockerfile:
```diff
@@ -0,0 +1,19 @@
+FROM openjdk:8-alpine
```
You should use a [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/) to build your maven package. That way the maven project will also be recompiled in the CI pipeline on new code changes.

Nicholas-Schaub commented:
I fixed the following things per requests of reviewers:
1. Made a multi-stage build file.
2. Added comments to pom.xml

In addition to this, I added the following features:
1. Instead of writing each plane separately, if there are z-slices present then volumes are saved in separate files by timepoint and channel. Filenames are changed so that if z-slices are present then the range of slices incorporated into a volume will be represented in the filename by `z<min-max>`. This is done to pre-emptively support changes that will be made for handling 3D data in WIPP.
2. Each image to be converted is given it's own thread for conversion.
3. This plugin originally checked file extensions of data in the image collection to make sure they were supported by Bioformats before conversion, and only file formats supported by Bioformats were converted to tiled tiff. However, Bioformats also supports conversion of plain text type files (txt, csv, xls/xlsx). This version of the plugin will not convert these file types to tiled tiff.
